### PR TITLE
App availability check for external share options

### PR DESCRIPTION
**Pull request by kumarpalsinh25**

### CODE REVIEW

File: app/lib/common/providers/app_install_check_provider.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      ExternalApps.signal => await checkAppAvailability(
+          Platform.isAndroid ? 'org.thoughtcrime.securesms' : 'sgnl://',
+        ),
+    };
+  }, // this means we are running
+);
+
+Future<bool> checkAppAvailability(String package) async {
+  try {
+    final app = await AppCheck().checkAvailability(package);
```
Comment (gnunicorn) [inline app/lib/common/providers/app_install_check_provider.dart:26]: Why not use [`isAppInstalled(package)`](https://pub.dev/documentation/appcheck/latest/appcheck/AppCheck/isAppInstalled.html)?

File: app/lib/common/providers/app_install_check_provider.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    return app != null;
+  } catch (e) {
```
Comment (gnunicorn) [inline app/lib/common/providers/app_install_check_provider.dart:28]: logging the error? What kind of error are we ever expecting here?

File: app/lib/common/widgets/share/widgets/external_share_options.dart [hunk 3]
```diff
@@ -23,9 +25,16 @@ class ExternalShareOptions extends StatelessWidget {
...
+  Widget build(BuildContext context, WidgetRef ref) {
     final lang = L10n.of(context);
 
+    final isWhatsAppInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.whatsApp)).valueOrNull;
+    final isTelegramInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.telegram)).valueOrNull;
+    final isSignalInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.signal)).valueOrNull;
+
```
Comment (gnunicorn) [inline app/lib/common/widgets/share/widgets/external_share_options.dart:37]: I'd move that `== true` check up here... just makes more sense to me ...

**Review by gnunicorn — APPROVED** [review]

File: app/lib/common/providers/app_install_check_provider.dart (hunk 1 above)
Comment (kumarpalsinh25) [reply app/lib/common/providers/app_install_check_provider.dart:26]: Ahh, good notice! Let me use it.

File: app/lib/common/providers/app_install_check_provider.dart (hunk 2 above)
Comment (kumarpalsinh25) [reply app/lib/common/providers/app_install_check_provider.dart:28]: If app is not install then `checkAvailability` function will give `PlatformException` so managed in this way.

Screenshot 2024-12-05 at 11 33 36 AM

But anyway, as per your suggestion, let me use `isAppInstalled` which is basically doing the same.
Screenshot 2024-12-05 at 11 34 58 AM

File: app/lib/common/widgets/share/widgets/external_share_options.dart (hunk 3 above)
Comment (kumarpalsinh25) [reply app/lib/common/widgets/share/widgets/external_share_options.dart]: Sure, let me change it accordingly.

**Pull request merged**